manager/api/transport_test.go

```diff
@@ -0,0 +1,535 @@
+			{"try to update client with invalid id", string(validUpdate), "1", validUser.Email, http.StatusNotFound},
+			{"try to update client with invalid data", string(invalidUpdate), clientID, validUser.Email, http.StatusBadRequest},
+			{"try to update client with invalid user token", string(validUpdate), clientID, invalidUser.Email, http.StatusForbidden},
+			{"try to update client with invalid data format", "{", clientID, validUser.Email, http.StatusBadRequest},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodPut, fmt.Sprintf("%s/clients/%s", url, tc.clientID), contentType, tc.auth, strings.NewReader(tc.req))
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+		}
+	})
+}
+
+func TestViewClient(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		clientID, _ := svc.AddClient(validUser.Email, validClient)
+
+		validClient.ID = clientID
+		validClient.Key = clientID
+		validData, _ := json.Marshal(validClient)
+
+		cases := []struct {
+			desc     string
+			clientID string
+			auth     string
+			status   int
+			res      string
+		}{
+			{"view existing client", clientID, validUser.Email, http.StatusOK, string(validData)},
+			{"try to view non-existent client", "123e4567-e89b-12d3-a456-000000000042", validUser.Email, http.StatusNotFound, ""},
+			{"try to view client by passing invalid id", "1", validUser.Email, http.StatusNotFound, ""},
+			{"try to view client by passing invalid token", clientID, invalidUser.Email, http.StatusForbidden, ""},
+		}
+
+		for _, tc := range cases {
+			res, err := sendRequest(client, http.MethodGet, fmt.Sprintf("%s/clients/%s", url, tc.clientID), "", tc.auth, nil)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			body, err := ioutil.ReadAll(res.Body)
+			assert.Nil(t, err, fmt.Sprintf("%s: unexpected error %s", tc.desc, err))
+			data := strings.Trim(string(body), "\n")
+			assert.Equal(t, tc.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", tc.desc, tc.status, res.StatusCode))
+			assert.Equal(t, tc.res, data, fmt.Sprintf("%s: expected body %s got %s", tc.desc, tc.res, data))
+		}
+	})
+}
+
+func TestListClients(t *testing.T) {
+	runService(func(svc manager.Service, url string, client *http.Client) {
+		svc.Register(validUser)
+		clients := []manager.Client{}
+		for i := 0; i < 10; i++ {
+			client := validClient
+			clientID, _ := svc.AddClient(validUser.Email, client)
+			client.ID = clientID
+			client.Key = clientID
+			clients = append(clients, client)
+		}
+
+		cases := []struct {
+			desc   string
+			auth   string
+			status int
+			res    []manager.Client
+		}{
+			{"fetch list of clients", validUser.Email, http.StatusOK, clients},
```

Comment:
Maybe add another case where we retrieve clients for user without any clients (you should create that user as well).